dashboard/lib/services/user/password_resetter.rb

```diff
@@ -10,14 +10,10 @@ def initialize(email:)
       def call
         return user if user.errors.present?
 
-        if user.authentication_options.any?(&:email?)
+        if user.authentication_options.any?(&:email?) || (!user.new_record? && !user.migrated? && user.authentication_options.empty?)
```

Comment:
```suggestion
        if user.authentication_options.any?(&:email?) || (user.provider == AuthenticationOption::EMAIL)
```
Would something like this work?

Reply:
I changed it to check if the provider is `manual` or `nil`. Looking through how we create "demigrated" users, it [looks like](https://github.com/code-dot-org/code-dot-org/blob/b9c34cd77eae59f3ebfe29c920b3b85aff1979b0/dashboard/lib/user_multi_auth_helper.rb#L99-L125) `[manual](https://github.com/code-dot-org/code-dot-org/blob/b9c34cd77eae59f3ebfe29c920b3b85aff1979b0/dashboard/app/models/user.rb#L200-L202)` and `nil` are the provider options are the ones that have a password login.